File: 05.01.插入/05.01.cpp
class Solution {
public:
    int insertBits(int N, int M, int i, int j) {
        for (int k = i; k <= j; k++) {
            if (N & (1 << k)) {
                N ^= (1 << k);
            }
        }

        M <<= i;
        return N + M;
    }
};

File: 05.01.插入/05.01.go
func insertBits(N int, M int, i int, j int) int {
	for k := i; k <= j; k++ {
		if (N & (1 << k)) > 0 {
			N = N ^ (1 << k)
		}
	}

	M = M << i
	return N + M
}